sensors/gps/pa6h.c

```diff
@@ -0,0 +1,204 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Driver for GPS module PA6H
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Damian Loewnau, Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <math.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/time.h>
+#include <sys/threads.h>
+
+#include "../sensors.h"
+#include "nmea.h"
+
+
+/* Specified by the module's documentation */
+#define UPDATE_RATE_MS 1000
+
+#define DEG2RAD     0.0174532925
+#define DEG2MILIRAD 17.4532925 /* 1 degree -> 1 milliradian conversion parameter */
+#define KMH2MMS     277.7778   /* 1 km/h -> mm/s conversion parameter */
+
+
+typedef struct {
+	sensor_event_t evtGps;
+	int filedes;
+	char buff[1024];
+	char stack[512] __attribute__((aligned(8)));
+} pa6h_ctx_t;
+
+
+int pa6h_update(nmea_t *message, pa6h_ctx_t *ctx)
+{
+	switch (message->type) {
+		case nmea_gga:
+			ctx->evtGps.gps.lat = message->msg.gga.lat * 1e7;
+			ctx->evtGps.gps.lon = message->msg.gga.lon * 1e7;
+			ctx->evtGps.gps.hdop = (unsigned int)(message->msg.gga.hdop * 1e2);
+			ctx->evtGps.gps.fix = message->msg.gga.fix;
+			ctx->evtGps.gps.alt = message->msg.gga.h_asl * 1e3;
+			ctx->evtGps.gps.altEllipsoid = message->msg.gga.h_wgs * 1e3;
+			ctx->evtGps.gps.satsNb = message->msg.gga.sats;
+			break;
+
+		case nmea_gsa:
+			ctx->evtGps.gps.hdop = (unsigned int)(message->msg.gsa.hdop * 1e2);
+			ctx->evtGps.gps.vdop = (unsigned int)(message->msg.gsa.vdop * 1e2);
+			break;
+
+		case nmea_rmc:
+			break;
+
+		case nmea_vtg:
+			ctx->evtGps.gps.heading = message->msg.vtg.track * DEG2MILIRAD;     /* degrees -> milliradians */
+			ctx->evtGps.gps.groundSpeed = message->msg.vtg.speed_kmh * KMH2MMS; /* kmh->mm/s */
+			ctx->evtGps.gps.velNorth = cos(message->msg.vtg.track * DEG2RAD) * ctx->evtGps.gps.groundSpeed;
+			ctx->evtGps.gps.velEast = sin(message->msg.vtg.track * DEG2RAD) * ctx->evtGps.gps.groundSpeed;
+			break;
+
+		default:
+			break;
+	}
+	gettime(&(ctx->evtGps.timestamp), NULL);
+
+	return 0;
+}
+
+
+static void pa6h_threadPublish(void *data)
+{
+	sensor_info_t *info = (sensor_info_t *)data;
+	pa6h_ctx_t *ctx = info->ctx;
+	nmea_t message;
+	char *start, **lines;
+	int i, nlines = 0, nbytes = 0, ret = 0;
+	unsigned char flag = 0;
```

Comment:
Maybe rename this `flag` variable to something more meaningful, like `doUpdate` or `updateFlag`.